lnbits/core/crud.py

```diff
@@ -114,17 +114,22 @@ def get_wallet(wallet_id: str) -> Optional[Wallet]:
 
 def get_wallet_for_key(key: str, key_type: str = "invoice") -> Optional[Wallet]:
     with open_db() as db:
-        check_field = "adminkey" if key_type == "admin" else "inkey"
         row = db.fetchone(
             f"""
             SELECT *, COALESCE((SELECT balance FROM balances WHERE wallet = wallets.id), 0) AS balance_msat
             FROM wallets
-            WHERE {check_field} = ?
+            WHERE adminkey = ? OR inkey = ?
             """,
-            (key,),
+            (key, key),
         )
 
-    return Wallet(**row) if row else None
+        if not row:
+            return None
+
+        if key_type in {"admin", "adminkey"} and row["adminkey"] != key:
```

Comment:
Maybe we just have to use "adminkey" and "invoicekey" as key_type if "admin" and "invoice" are confusing, instead of allowing all options. We can easily search and replace the old values in the decorators.

Reply:
Ok. I was just confused about keys having one name and we having to specify a different name here as `key_type`.